pkg/host/internal/infiniband/infiniband.go

```diff
@@ -0,0 +1,67 @@
+package infiniband
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"net"
+
+	"github.com/vishvananda/netlink"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/consts"
+	netlinkLibPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/types"
+)
+
+// New creates and returns an InfinibandInterface object, that handles IB VF GUID configuration
+func New(netlinkLib netlinkLibPkg.NetlinkLib, kernelHelper types.KernelInterface, networkHelper types.NetworkInterface) (types.InfinibandInterface, error) {
+	guidPool, err := newIbGUIDPool(consts.InfinibandGUIDConfigFilePath, netlinkLib, networkHelper)
```

Comment:
be aware here is better to use utils.GetHostExtension to join the InfinibandGUIDConfigFilePath in case we are not inside chroot the function will not work

Reply:
GetHostExtensionPath is used at the top of the call stack in [`ib_guid_config.go:98`](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/a718c3dc7113a4def3b03bf09123b5b8259d60a9/pkg/host/internal/infiniband/ib_guid_config.go#L98)

Reply:
@almaslennikov i think GetHostExtensionPath need to be taken into account here as well according to @SchSeba 
like in https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/18c46954d3998f3b9450db7f4fdee4d687759923/pkg/host/store/store.go#L46